modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -784,7 +784,8 @@ private CPDefinition _updateNestedResources(
 
 		// Product configuration
 
-		ProductConfiguration productConfiguration = product.getConfiguration();
+		ProductConfiguration productConfiguration =
+			product.getProductConfiguration();
 
```

Comment:
you should also remove the underneath if

Reply:
most probably

Reply:
Hi @andrea-ale-sbarra, could you explain to me why should this be removed? I took a look and every other nested resource has a similar if. As i understand this is just to update the CpDefinitionInventory with the new product configuration.

Reply:
Ehi @victorcdp forget about that I was mislead by  the getter methd that you changed I thought the underneath if was something old to change too, but it was not.